subsys/net/ip/net_context.c

```diff
@@ -689,6 +689,45 @@ int net_context_put(struct net_context *context)
 	return 0;
 }
 
+static int net_context_bind_default(struct net_context *context,
+				    const struct sockaddr *dest_addr)
```

Comment:
Yes, dest_addr is passed so we can select address family. My initial idea was just to a special case to existing net_context_bind(), when addr=NULL is passed to it, perform this default binding. But then we wouldn't know with AF to select indeed. So instead, I factored out this logic into separate function, which accepts dest address. It's called from connect(), where destaddr is known, and I may imagine it may need to be called form recv(), send() on UDP path, where destaddr is also known.

So, that's the reason why this function is written like it is - it mimics net_context_bind(). Note that's not the only way to select AF, because context is created with specific AF ;-). But doing it like bind() does kinda leave possibility open to have Linux-style dual-family sockets.